cache.go

```diff
@@ -371,11 +361,14 @@ func (c *Cache[K, V]) Items() map[K]*Item[K, V] {
 }
 
 // Metrics returns the metrics of the cache.
-func (c *Cache[K, V]) Metrics() Metrics {
-	c.metricsMu.RLock()
-	defer c.metricsMu.RUnlock()
-
-	return c.metrics
+func (c *Cache[K, V]) Metrics() (m Metrics) {
+	// I imagine on most architectures this is equivalent to
```

Comment:
Just curious, what do you mean by this comment? Afaik it's not the best idea to do just `return c.metrics` on any platforms.

Reply:
I submitted this PR almost 2 months ago and no longer remember the details of the code. What I vaguely remember is that it seemed like there may unnecessary atomic operations / locks in various places. Granted, some architectures may require atomic loads where others don't. However, I don't have time to re-read all of the code again now. I'm guessing that what I was indicating in that comment is that the atomic loads are of no value if they're already guarded by a read lock -- if they are, then returning a struct by value (plus or minus any golang quirks) is likely pretty equivalent to creating a new struct and copying the values over explicitly in your code.

Anyway, all of the above caveat emptor because it's been too long for me to remember.